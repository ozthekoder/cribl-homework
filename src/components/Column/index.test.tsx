import { render } from "@testing-library/react";
import { Column } from ".";
import "@testing-library/jest-dom";

describe("Column Component", () => {
  test("should render correctly with default styles", () => {
    const { container } = render(<Column>Test Content</Column>);
    expect(container.firstChild).toHaveStyle({
      display: "flex",
      flexDirection: "column",
      justifyContent: "flex-start",
      width: "100%",
    });
  });

  test("should apply width correctly when passed as a prop", () => {
    const { container } = render(<Column width="200px">Test Content</Column>);
    expect(container.firstChild).toHaveStyle({
      width: "200px",
    });
  });

  test("should render children correctly", () => {
    const { getByText } = render(<Column>Child Element</Column>);
    expect(getByText("Child Element")).toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { container } = render(<Column width="300px">Snapshot Test</Column>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
