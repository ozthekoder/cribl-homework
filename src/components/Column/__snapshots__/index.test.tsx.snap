// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Component should match snapshot 1`] = `
<div
  class="css-c0d617"
  width="300px"
>
  Snapshot Test
</div>
`;
